apps/ws/src/shared/framework/in-memory-io.adapter.ts

```diff
@@ -0,0 +1,38 @@
+import { IoAdapter } from '@nestjs/platform-socket.io';
+import { ServerOptions } from 'socket.io';
+import Redis from 'ioredis';
+import { createAdapter } from '@socket.io/redis-adapter';
+import { getRedisPrefix } from '@novu/shared';
+import { WebSocketsInMemoryProviderService } from '@novu/application-generic';
+import { ConnectionOptions } from 'tls';
+
+export class InMemoryIoAdapter extends IoAdapter {
+  private webSocketsInMemoryProviderService: WebSocketsInMemoryProviderService;
+  private adapterConstructor: ReturnType<typeof createAdapter>;
+
+  async connectToInMemoryCluster(): Promise<void> {
+    // TODO: Pending to inject in the provider instantiation
+    const keyPrefix = getRedisPrefix() ? `socket.io#${getRedisPrefix()}` : 'socket.io';
```

Comment:
do you plan to make this change in a separate PR?

Reply:
This change is dependant and also this PR on this other PR: https://github.com/novuhq/novu/pull/4868
I gave the full context in the Linear task.

Reply:
ok, thanks, will take a look into that PR ;)

Reply:
maybe then rebase this one to `nv-3197-move-provider-selection-to-workflow-and-cache-in-memory`?

Reply:
Yes, I will do that once NV-3197 is reviewed and approved.